showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/WickedClientTest.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.showcase.v1beta1;
+
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.grpc.GaxGrpcProperties;
+import com.google.api.gax.grpc.testing.LocalChannelProvider;
+import com.google.api.gax.grpc.testing.MockGrpcService;
+import com.google.api.gax.grpc.testing.MockServiceHelper;
+import com.google.api.gax.grpc.testing.MockStreamObserver;
+import com.google.api.gax.rpc.ApiClientHeaderProvider;
+import com.google.api.gax.rpc.ApiStreamObserver;
+import com.google.api.gax.rpc.BidiStreamingCallable;
+import com.google.api.gax.rpc.ClientStreamingCallable;
+import com.google.api.gax.rpc.InvalidArgumentException;
+import com.google.api.gax.rpc.StatusCode;
+import com.google.protobuf.AbstractMessage;
+import io.grpc.StatusRuntimeException;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import javax.annotation.Generated;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+@Generated("by gapic-generator-java")
+public class WickedClientTest {
```

Comment:
Is there any manual showcase tests added? It's great that you added a new showcase proto, but if there is no interactions with the showcase server, I would suggest to use the unit tests or golden integration tests for simplicity.

Reply:
@blakeli0 : I requested we go this route to be explicit about what the golden tests are verifying with the goal to eventually decouple ourselves from specific APIs that may change over time outside of our control. If a service today only supports gRPC but later supports REST we will have lost our golden test for this behavior.

Reply:
@blakeli0 I'll forward you the email where we had some discussions with Noah and Victor. I added a `showcase-extended` module to hold some non-api specific cases that aren't covered by the existing showcase test cases.

Reply:
> If a service today only supports gRPC but later supports REST we will have lost our golden test for this behavior.

Yes that's a good test case. However, this test case does not require us to interact with showcase server, and we can achieve the same thing in unit golden tests. Currently we have two identical `wicked.proto` in both unit test proto and showcase proto, and they are testing the same thing from what I can see.

Reply:
I agree it might be a bit overkill to have this multiple times. The `wicked.proto` in unit tests was duplicated to help ensure sonarcode coverage (as none of the existing unit tests protos covered this path). `apigee connect v1` was used as an existing library for the (legacy?) golden ITs and the showcase `wicked.proto` was built as a initial framework for expanding showcase coverage (being able to test for generated client behavior and generator generation behavior).

@burkedavison @blakeli0  Do you think it would make sense to just include this as part of the unit tests proto and not showcase proto?

Reply:
I agree they only need to be in one place. No strong opinion on which.

Reply:
> I would eventually like to move away from golden ITs

I agree. But I think we should have the majority test cases in unit tests(golden or non-golden) instead of in showcase tests. Showcase tests are a great replacement for the integration tests that interact with servers, but they are still more expensive than unit tests.
> I agree they only need to be in one place. No strong opinion on which.

Thanks Burke! Also thanks for helping Lawrence setting up showcase tests for this extended proto, even though we may not use it this time, but we will definitely need it for the upcoming Dynamic Routing Header features.

Reply:
Understood, I can leave the framework for extending showcase (keeping the showcase-extendend module and the related bazel build files). I'll use the golden units for now with `wicked.proto`.